### PR TITLE
Bug/fix slow updates

### PR DESCRIPTION
**Pull request by Secretmapper**
PR fixes a bug where any/all changes to the torrent-store would not cause rpc to refetch instantly, causing the store/cache to update only on interval refetches.

Note that this is different from https://github.com/fcsonline/react-transmission/issues/35, which aims tackle perceived performance only.

### CODE REVIEW

File: src/stores/torrent-store.js [hunk 1]
```diff
@@ -57,85 +57,60 @@ class TorrentStore {
...
-        );
+        const newTorrents = result.arguments.torrents;
+
+        if (torrentIds) {
+          // There are ids passed to fetch, edit cache of dirty torrents
+          // XXX: This is a potentially slow operation. Perhaps cache should be a map?
+          newTorrents.forEach(torrent => {
+            const i = this.torrents.findIndex(t => t.id === torrent.id);
+            if (i >= 0)
+              this.torrents[i] = new Torrent(torrent);
```
Comment (edulan) [inline src/stores/torrent-store.js]: Maybe would be interesting to implement a `set` method in `Torrent` class. That way we could reuse the same instance and we wouldn't need to recreate models again and again. What do you think @Secretmapper?

Comment (Secretmapper) [reply src/stores/torrent-store.js]: I agree. Setting the properties seems to be the way to go for MobX.

**Comment by Secretmapper** [comment]
@edulan, I'm updating the props now. Not sure how much overhead extendObservable adds for existing properties, though at least we're sure it's not creating a new Torrent instance.

**Comment by fcsonline** [comment]
Merging and fixing errors later

**Pull request merged**